src/core.ts

```diff
@@ -324,19 +324,23 @@ function SqlPouch(opts: OpenDatabaseOptions, cb: (err: any) => void) {
       sqlArgs = [id, opts.rev]
     }
 
-    tx.executeAsync(sql, sqlArgs).then((results) => {
+    tx.execute(sql, sqlArgs).then((results) => {
       if (!results.rows?.length) {
         const missingErr = createError(MISSING_DOC, 'missing')
         return finish(missingErr)
       }
-      const item = results.rows.item(0)
+      const item = results.rows[0]!
       metadata = safeJsonParse(item.metadata)
       if (item.deleted && !opts.rev) {
         const deletedErr = createError(MISSING_DOC, 'deleted')
         return finish(deletedErr)
       }
-      doc = unstringifyDoc(item.data, metadata.id, item.rev)
+      doc = unstringifyDoc(item.data as string, metadata.id, item.rev as string)
       finish(null)
+    }).catch(e => {
```

Comment:
Can you revert this 'catching all errors' part?

Reply:
No, this catch is important, since it rejects the promise of async execution. If removed, pouchdb.get() will remain unresolved.

Reply:
You mean, it is for catching errors thrown from `tx.execute`?

Reply:
Yes, or `tx.execute` will hang